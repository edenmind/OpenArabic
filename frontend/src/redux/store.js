import {
  arabicSentenceReducer,
  arabicTextReducer,
  arabicWordsReducer,
  authorReducer,
  categoryReducer,
  englishSentenceReducer,
  englishTextReducer,
  englishWordsReducer,
  sourceReducer,
  textReducer,
  wordByWordReducer,
} from './reducers'

import { configureStore } from '@reduxjs/toolkit'

export default configureStore({
  reducer: {
    title: textReducer,
    category: categoryReducer,
    author: authorReducer,
    source: sourceReducer,
    englishText: englishTextReducer,
    arabicText: arabicTextReducer,
    arabicWords: arabicWordsReducer,
    englishWords: englishWordsReducer,
    arabicSentence: arabicSentenceReducer,
    englishSentence: englishSentenceReducer,
    wordByWord: wordByWordReducer,
  },
})
