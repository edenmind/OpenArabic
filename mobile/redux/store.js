import { configureStore, combineReducers } from '@reduxjs/toolkit'

import { categoriesReducer, textReducer, textsReducer, UIStateReducer, wordsReducer } from './reducers.js'

const rootReducer = combineReducers({
  categories: categoriesReducer,
  text: textReducer,
  texts: textsReducer,
  textLoading: UIStateReducer,
  textsLoading: UIStateReducer,
  arabicFontSize: UIStateReducer,
  englishFontSize: UIStateReducer,
  isTransliterationOn: UIStateReducer,
  arabicFontName: UIStateReducer,
  practicingWords: UIStateReducer,
  words: wordsReducer,
  isDarkMode: UIStateReducer
})

const store = configureStore({
  reducer: rootReducer
})

export { store }
