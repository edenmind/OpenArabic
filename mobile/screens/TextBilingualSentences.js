/* eslint-disable import/namespace */
import * as util from '../services/UtilityService'

import { Button, Text } from 'react-native-paper'
import React, { Fragment } from 'react'
import { StyleSheet, View } from 'react-native'

import { COLORS } from '../constants/colors'
import ModalScrollView from '../components/ModalScrollView'
import PropTypes from 'prop-types'
import WordPairs from './TextBilingualSentencesWordPairs'

const style = StyleSheet.create({
  arabic: {
    direction: 'rtl',
    flex: 1,
    fontSize: 25,
    lineHeight: 33,
    paddingBottom: 10,
    paddingLeft: 33,
    paddingRight: 33,
    paddingTop: 33,
    writingDirection: 'rtl'
  },
  bottomPadding: {
    paddingBottom: 55
  },
  english: {
    direction: 'ltr',
    flex: 1,
    lineHeight: 25,
    paddingLeft: 33,
    paddingRight: 33,
    writingDirection: 'ltr'
  },
  showWordsButton: {
    paddingLeft: 100,
    paddingRight: 100
  },
  showWordsText: {
    color: COLORS.darkGrey
  }
})

export default function TextBilingualSentences(props) {
  const [visible, setVisible] = React.useState(false)
  const [words, setWords] = React.useState([])
  const hideModal = () => setVisible(false)
  const showModal = () => setVisible(true)
  const getListOfWordPairs = (index) => setWords(index)

  const filterFunction = (element) => element.english !== '' && element.arabic !== ''

  const sentences = props.sentences.map((sentence, index) => (
    <Fragment key={index}>
      <Text style={style.arabic}>{sentence.arabic}</Text>
      <Text style={style.english}>{sentence.english}</Text>

      <Button
        style={style.showWordsButton}
        mode="text"
        onPress={() => {
          getListOfWordPairs(
            <WordPairs words={util.filterArrayFromEmptyElements(sentence.words, filterFunction)} />
          )
          showModal()
        }}>
        <Text style={style.showWordsText}>Vocabulary</Text>
      </Button>
    </Fragment>
  ))

  return (
    <Fragment>
      <View style={style.bottomPadding}>{sentences}</View>
      <ModalScrollView visible={visible} content={words} title="Vocabulary" hideModal={hideModal} />
    </Fragment>
  )
}

TextBilingualSentences.propTypes = {
  sentences: PropTypes.arrayOf(
    PropTypes.shape({
      word: PropTypes.string,
      wordId: PropTypes.string
    })
  )
}
