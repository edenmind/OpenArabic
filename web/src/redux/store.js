import * as reducers from './reducers'

import { configureStore } from '@reduxjs/toolkit'

export default configureStore({
  reducer: {
    text: reducers.textReducer,
  },
})
