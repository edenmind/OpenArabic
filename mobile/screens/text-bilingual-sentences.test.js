import { describe, it, jest } from '@jest/globals'
import { Provider } from 'react-redux'
import { store } from '../redux/store.js'
import React from 'react'
import TextBilingualSentences from './text-bilingual-sentences.js'
import { NavigationContainer } from '@react-navigation/native'
import ShallowRenderer from 'react-test-renderer/shallow' // ES6

jest.useFakeTimers()
describe('<Sentence />', () => {
  it('renders without crashing', () => {
    // eslint-disable-next-line putout/objects-braces-inside-array
    const propsMock = [
      {
        sentenceID: '1',
        arabic: [{ a: 'a' }],
        english: [{ e: 'e' }],
        sentenceId: 1
      }
    ]
    // @ts-ignore
    const renderer = new ShallowRenderer()
    const tree = renderer.render(
      <Provider store={store}>
        <NavigationContainer>
          <TextBilingualSentences sentences={propsMock} />
        </NavigationContainer>
      </Provider>
    )

    expect(tree).toMatchSnapshot()
  })
})
