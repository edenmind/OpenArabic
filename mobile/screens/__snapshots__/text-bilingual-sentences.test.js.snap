// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sentence /> renders without crashing 1`] = `
<Context.Provider
  value={
    {
      "getServerState": undefined,
      "store": {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <ForwardRef(NavigationContainerInner)>
    <TextBilingualSentences
      sentences={
        [
          {
            "arabic": [
              {
                "a": "a",
              },
            ],
            "english": [
              {
                "e": "e",
              },
            ],
            "sentenceID": "1",
            "sentenceId": 1,
          },
        ]
      }
    />
  </ForwardRef(NavigationContainerInner)>
</Context.Provider>
`;
